squad_client/shortcuts.py

```diff
@@ -24,12 +25,57 @@ def retrieve_build_results(build_url):
             testrun = testruns[_id]
             test_suites = {}
             for suite in testrun.test_suites:
-                test_suites[suite.name] = {t.short_name: t.status for t in suite.tests}
+                test_suites[suite.name] = {t.short_name: t.status for t in suite.tests.values()}
 
             metric_suites = {}
             for suite in testrun.metric_suites:
-                metric_suites[suite.name] = {m.name: m.result for m in suite.metrics}
+                metric_suites[suite.name] = {m.name: m.result for m in suite.metrics.values()}
 
             results[testrun.environment.slug] = {'tests': test_suites, 'metrics': metric_suites}
 
     return results
+
+
+def submit_results(group_project_slug=None, build_version=None, env_slug=None, tests={}, metrics={}, log=None, metadata=None, attachments=None):
+    group_slug, project_slug = split_group_project_slug(group_project_slug)
+
+    # TODO: validate input
+
+    group = Group()
+    project = Project()
+    build = Build()
+    testrun = TestRun()
+    environment = Environment()
+
+    group.slug = group_slug
+    project.group = group
+    project.slug = project_slug
+    environment.slug = env_slug
+    environment.project = project
+    build.project = project
+    build.version = build_version
+
+    testrun.log = log
+    testrun.build = build
+    testrun.metadata = metadata
+    testrun.attachments = attachments
+    testrun.environment = environment
+
+    for test_name in tests.keys():
+        test = Test()
+        test.name = test_name
+        testrun.add_test(test)
+
+        if type(tests[test_name]) is dict:
+            test.status = tests[test_name].get('result')  # raise error if result is invalid
```

Comment:
I'm adding a check function in command. Maybe it should be shared so it can be reused here as well?

Reply:
I like that, but maybe the function should live in utils?

Reply:
Yes, you're 100% correct. That is my plan.